### PR TITLE
Opinion model

### PR DESCRIPTION
**Pull request by korneliacieslik**

### CODE REVIEW

File: app/views/opinions/index.html.erb [hunk 1]
```diff
@@ -1,13 +1,17 @@
 <h1>Your opinions about us</h1>
 
 <button type="button", class="nav-button btn-link">
-	
-	<%= link_to "Add your opinion", opinions_path %>
```
Comment (korneliacieslik) [inline app/views/opinions/index.html.erb]: Doesn't create link.

File: test/controllers/opinion_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require 'test_helper'
+
+class OpinionControllerTest < ActionDispatch::IntegrationTest
```
Comment (rodzyn) [inline test/controllers/opinion_controller_test.rb]: Do you think that those tests are needed? We are testing here built-in rails helpers and assert typical responses. All those is quite well tested in rails itself

File: db/migrate/20200116203450_add_timestamps_to_opinions.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class AddTimestampsToOpinions < ActiveRecord::Migration[6.0]
+  def change
+  # add new column but allow null values
```
Comment (rodzyn) [inline db/migrate/20200116203450_add_timestamps_to_opinions.rb]: You can do:
`
  def change
     add_timestamps : opinions, null: false, default: -> { 'NOW()' }
  end
`
You should probably avoid direct manipulation on db like with `update_all` - a typo could make disaster ;)

File: config/routes.rb [hunk 4]
```diff
@@ -1,11 +1,19 @@
...
+Rails.application.routes.draw do 
+  resources :customers, :opinions, :orders, :products
+
+  get 'opinions/index'
+  get 'opinions/new'
+  post'opinions/create'
+  get 'opinions/destroy'
+  get 'opinions/edit'
+  post'opinions/update'
+
```
Comment (rodzyn) [inline config/routes.rb]: All those, probably not needed because we have `resources :opinions`

File: app/views/opinions/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<h1>Add opinion</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+  	<%= form_for  :opinion do |f| %>
+  		<%= f.text_area :content %>
+  		<%= f.submit %>
+  	<% end %>
```
Comment (rodzyn) [inline app/views/opinions/new.html.erb]: As we agreed this should be more like in the product:

`
<%= render 'form', opinion: @opinion %>
`

That's the way to render `_form.html.erb` that you have added in this PR

`@opinion` is an instance variable that we are passing from `OpinionsController#new` method

File: test/controllers/opinion_controller_test.rb (hunk 2 above)
Comment (korneliacieslik) [reply test/controllers/opinion_controller_test.rb]: Ok, I understand. :) So should I remove that code?

File: db/migrate/20200116203450_add_timestamps_to_opinions.rb (hunk 3 above)
Comment (korneliacieslik) [reply db/migrate/20200116203450_add_timestamps_to_opinions.rb]: I changed it

File: config/routes.rb (hunk 4 above)
Comment (korneliacieslik) [reply config/routes.rb]: Yes! I remember and I overlooked it! 
It's removed now :)

File: app/views/opinions/new.html.erb (hunk 5 above)
Comment (korneliacieslik) [reply app/views/opinions/new.html.erb]: _form.html.erb is added

File: test/controllers/opinion_controller_test.rb (hunk 2 above)
Comment (rodzyn) [reply test/controllers/opinion_controller_test.rb]: It's up to you! :)

File: db/migrate/20200116203450_add_timestamps_to_opinions.rb [hunk 6]
```diff
@@ -1,7 +1,7 @@
 class AddTimestampsToOpinions < ActiveRecord::Migration[6.0]
   def change
   # add new column but allow null values
-  add_timestamps :opinions, null: true 
+  add_timestamps :opinions, null: false, defeault: -> { 'NOW()' }
 
   # backfill existing record with created_at and updated_at
   # values making clear that the records are faked
```
Comment (rodzyn) [inline db/migrate/20200116203450_add_timestamps_to_opinions.rb]: You can remove the code below now ;)

**Comment by rodzyn** [comment]
Let's merge that after you apply all the remaining fixes! ;)

**Pull request merged**